core/src/test/java/io/questdb/griffin/AlterTableAttachPartitionFromSoftLinkTest.java

```diff
@@ -0,0 +1,998 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin;
+
+import io.questdb.cairo.*;
+import io.questdb.cairo.security.AllowAllCairoSecurityContext;
+import io.questdb.std.*;
+import io.questdb.std.datetime.microtime.TimestampFormatUtils;
+import io.questdb.tasks.ColumnPurgeTask;
+import io.questdb.test.tools.TestUtils;
+import org.junit.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+
+public class AlterTableAttachPartitionFromSoftLinkTest extends AlterTableAttachPartitionBase {
```

Comment:
There should be a way to test all of this without relying on symbol link.

For example there could be a test-only method, which will make partition read only. Then relevant tests will pass on all platforms

Reply:
??? these tests are based on user use cases, not on artificial usage of backdoor method. 
I will soon be able to review the tests on a windows box, it may be as simple as releasing inactive readers/writers. I would prefer to leave this as is and readdress at a later time, given there are no users in windows, if that is ok.